src/main/java/graphql/schema/GraphQLAppliedDirective.java

```diff
@@ -0,0 +1,247 @@
+package graphql.schema;
+
+
+import com.google.common.collect.ImmutableList;
+import graphql.PublicApi;
+import graphql.language.Directive;
+import graphql.util.TraversalControl;
+import graphql.util.TraverserContext;
+
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.StringJoiner;
+import java.util.function.Consumer;
+import java.util.function.UnaryOperator;
+
+import static graphql.Assert.assertNotNull;
+import static graphql.Assert.assertValidName;
+import static graphql.util.FpKit.getByName;
+
+/**
+ * An applied directive repreents the instance of a  directive that is applied to a schema element,
+ * as opposed to it definition
+ * <p>
+ * See http://graphql.org/learn/queries/#directives for more details on the concept.
+ */
+@PublicApi
+public class GraphQLAppliedDirective implements GraphQLNamedSchemaElement {
+
+    private final String name;
+    private final ImmutableList<GraphQLAppliedDirectiveArgument> arguments;
+    private final Directive definition;
+
+    public static final String CHILD_ARGUMENTS = "arguments";
+
+    private GraphQLAppliedDirective(String name, Directive definition, List<GraphQLAppliedDirectiveArgument> arguments) {
+        assertValidName(name);
+        assertNotNull(arguments, () -> "arguments can't be null");
+        this.name = name;
+        this.arguments = ImmutableList.copyOf(arguments);
+        this.definition = definition;
+    }
+
+    @Override
+    public String getName() {
+        return name;
+    }
+
+    @Override
```

Comment:
does an applied directive actually have a description? Not 100% sure out of my head.

Reply:
They do no have descriptions and the grammar does not allow it

However out interfaces call for it

```
public interface GraphQLNamedSchemaElement extends GraphQLSchemaElement {
    String getName();
    String getDescription();
    Node getDefinition();
}
```

Reply:
probably should add Nullable then to it or add a comment

Reply:
done